demisto_sdk/commands/download/downloader.py

```diff
@@ -748,520 +985,487 @@ def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
             return None
-        return playbooks[0]["id"]
+
+        playbook_id = playbooks[0]["id"]
+        logger.info(f"Found playbook ID '{playbook_id}' for '{playbook_name}'")
+        return playbook_id
 
     @staticmethod
-    def get_main_file_details(content_entity: str, entity_instance_path: str) -> tuple:
+    def get_metadata_file(content_type: str, content_path: Path) -> dict | None:
         """
-        Returns the details of the "main" file within an entity instance.
-        For example: In the HelloWorld integration under Packs/HelloWorld, the main file is the yml file.
-        It contains all relevant ids and names for all the files under the HelloWorld integration dir.
-        :param content_entity: The content entity, for example Integrations
-        :param entity_instance_path: For example: ~/.../content/Packs/TestPack/Integrations/HelloWorld
-        :return: The main file id & name
-        """
-        main_file_data: dict = dict()
-        main_file_path: str = ""
+        Returns the data of the "main" file containing metadata for a content item's path.
+        For example, YAML file for integrations / scripts, playbook file for playbooks, etc.
+
+        Args:
+            content_type (str): The type of the content item.
+            content_path (Path): The path to the content item.
 
-        # Entities which contain yml files
-        if content_entity in (
+        Returns:
+            dict | None: The data of the "main" file. None if the file could not be found / parsed.
+        """
+        if content_type in (
             INTEGRATIONS_DIR,
             SCRIPTS_DIR,
             PLAYBOOKS_DIR,
             TEST_PLAYBOOKS_DIR,
         ):
-            if os.path.isdir(entity_instance_path):
-                _, main_file_path = get_yml_paths_in_dir(entity_instance_path)
-            elif Path(entity_instance_path).is_file():
-                main_file_path = entity_instance_path
+            if content_path.is_dir():
+                main_file_path = get_yml_paths_in_dir(content_path)[1]
 
-            if main_file_path:
-                main_file_data = get_yaml(main_file_path)
+                if not main_file_path:
+                    return None
 
-        # Entities which are json files (md files are ignored - changelog/readme)
-        else:
-            if (
-                Path(entity_instance_path).is_file()
-                and retrieve_file_ending(entity_instance_path) == "json"
-            ):
-                main_file_data = get_json(entity_instance_path)
+                return get_yaml(main_file_path)
+
+            elif content_path.is_file():
+                return get_yaml(content_path)
 
-        main_id = get_entity_id_by_entity_type(main_file_data, content_entity)
-        main_name = get_entity_name_by_entity_type(main_file_data, content_entity)
+        else:
+            if content_path.is_file() and content_path.suffix == ".json":
```

Comment:
```suggestion
            if content_item_path.is_file() and content_path.suffix == ".json":
```
`content_path` sounds like the root of the content repo which makes it unclear